tests/meta/demultiplex/test_validation.py

```diff
@@ -172,22 +173,29 @@ def test_is_flow_cell_ready_for_delivery_false(tmp_path: Path):
     assert result == False
 
 
-def test_validate_sample_sheet_exists_raises_error(tmp_path: Path):
-    # GIVEN a path with no sample sheet
-
+def test_validate_sample_sheet_exists_raises_error(bcl2fastq_flow_cell_dir: Path):
+    # GIVEN a flow cell without a sample sheet in housekeeper
+    flow_cell = FlowCellDirectoryData(flow_cell_path=bcl2fastq_flow_cell_dir)
+    flow_cell._sample_sheet_path_hk = None
     # WHEN validating the existence of the sample sheet
     # THEN it should raise a FlowCellError
     with pytest.raises(FlowCellError):
-        validate_sample_sheet_exists(tmp_path)
+        validate_sample_sheet_exists(flow_cell=flow_cell)
 
 
-def test_validate_sample_sheet_exists_no_error(tmp_path: Path):
+def test_validate_sample_sheet_exists_no_error(bcl2fastq_flow_cell_dir: Path):
     # GIVEN a path with a sample sheet
-    (tmp_path / DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME).touch()
+    # GIVEN a flow cell without a sample sheet in housekeeper
+    flow_cell = FlowCellDirectoryData(flow_cell_path=bcl2fastq_flow_cell_dir)
+    sample_sheet_path = Path(
+        bcl2fastq_flow_cell_dir, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME
+    )
+    sample_sheet_path.touch()
+    flow_cell._sample_sheet_path_hk = sample_sheet_path
 
     # WHEN validating the existence of the sample sheet
     # THEN it should not raise an error
-    assert validate_sample_sheet_exists(tmp_path) is None
+    assert validate_sample_sheet_exists(flow_cell=flow_cell) is None
```

Comment:
Sort of similar to above although "validate_sample_sheet_exists" is perhaps a bit better. Still feels like "assert is None" should be a failure though, and I would probably prefer a boolean return type on validate_sample_sheet_exists.

Reply:
A bit unsure though. I could be convinced that validate functions should be void and only raise an exception if anything goes wrong.

Reply:
I agree, it was a bit confusing when I looked at it as well. @diitaz93 and I were talking about a demux post processing refactoring when we established the new flow. We will have a look at it then.

Reply:
This could be discussed in coding guidelines meetings: return type of validators. I think for this one is ok to return None. The one above should return bool (`is_flow_cell_ready_for_postprocessing`)

Reply:
In line with the comment above. I also disagree with the return type of the function. I will leave it for now though.